pkg/kv/kvserver/replica_rangefeed.go

```diff
@@ -378,13 +375,19 @@ func (r *Replica) registerWithRangefeedRaftMuLocked(
 	ctx context.Context,
 	span roachpb.RSpan,
 	startTS hlc.Timestamp, // exclusive
-	catchUpIter rangefeed.CatchUpIteratorConstructor,
+	catchUpIter *rangefeed.CatchUpIterator,
 	withDiff bool,
 	stream rangefeed.Stream,
 	done *future.ErrorFuture,
 ) rangefeed.Processor {
 	defer logSlowRangefeedRegistration(ctx)()
 
+	cleanupCatchUpIter := func() {
+		if catchUpIter != nil {
+			catchUpIter.Close()
+		}
+	}
```

Comment:
We could consider inverting the logic here, using `defer` for this check and instead set `catchUpIter = nil` when we successfully register. That way, we avoid messing up any future error return paths we may add later.